cranelift/codegen/src/isa/riscv64/mod.rs

```diff
@@ -219,10 +219,34 @@ pub fn isa_builder(triple: Triple) -> IsaBuilder {
     IsaBuilder {
         triple,
         setup: riscv_settings::builder(),
-        constructor: |triple, shared_flags, builder| {
-            let isa_flags = riscv_settings::Flags::new(&shared_flags, builder);
-            let backend = Riscv64Backend::new_with_flags(triple, shared_flags, isa_flags);
-            Ok(backend.wrapped())
-        },
+        constructor: isa_constructor,
     }
 }
+
+fn isa_constructor(
+    triple: Triple,
+    shared_flags: Flags,
+    builder: &shared_settings::Builder,
+) -> CodegenResult<OwnedTargetIsa> {
+    let isa_flags = riscv_settings::Flags::new(&shared_flags, builder);
+
+    // The RISC-V backend does not work without at least the G extension enabled.
+    // The G extension is simply a combination of the following extensions:
+    // - I: Base Integer Instruction Set
+    // - M: Integer Multiplication and Division
+    // - A: Atomic Instructions
+    // - F: Single-Precision Floating-Point
+    // - D: Double-Precision Floating-Point
+    // - Zicsr: Control and Status Register Instructions
+    // - Zifencei: Instruction-Fetch Fence
+    //
+    // Ensure that those combination of features is enabled.
+    if !isa_flags.has_g() {
+        return Err(CodegenError::Unsupported(
+            "The RISC-V Backend requires all the features in the G Extension enabled".into(),
```

Comment:
```suggestion
            "The RISC-V Backend currently requires all the features in the G Extension enabled".into(),
```

In the future it should be possible to add support for more restricted feature sets, right?

Reply:
Oh yeah, we should be able to reduce, ideally all the way until RV64I which is just the 37 base instructions!